### PR TITLE
Fix router deleting

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

File: akanda/rug/notifications.py [hunk 1]
```diff
@@ -73,6 +74,7 @@ def _make_event_from_message(message):
             crud = event.CREATE
         elif event_type == 'router.delete.end':
             crud = event.DELETE
+            router_id = message.get('payload', {}).get('router_id')
```
Comment (ryanpetrello) [inline akanda/rug/notifications.py:77]: Is the router ID just stored differently for this type of event?  Care to elaborate on this a bit?

Comment (dhellmann) [inline akanda/rug/notifications.py:77]: Apparently neutron's notifications aren't consistent. Who would have guessed?

File: akanda/rug/state.py [hunk 2]
```diff
@@ -229,30 +234,26 @@ def __init__(self, router_id, tenant_id,
...
         "Called when the parent process is being stopped"
 
     def _do_delete(self):
         if self._delete_callback is not None:
+            self.log.debug('calling delete callback')
             self._delete_callback()
             # Avoid calling the delete callback more than once.
             self._delete_callback = None
+            # Remember that this router has been deleted
+            self.deleted = True
```
Comment (ryanpetrello) [inline akanda/rug/state.py]: This should always be set to True in this method even if `_delete_callback` _is_ None

Comment (dhellmann) [inline akanda/rug/state.py]: Fixed.

**Pull request merged**